processing/src/test/java/io/druid/segment/filter/JavaScriptDimFilterTest.java

```diff
@@ -0,0 +1,273 @@
+    {
+      return factory;
+    }
+
+    @Override
+    public ImmutableBitmap getBitmapIndex(String dimension, String value)
+    {
+      return bitmapMap.get(dimension).get(value);
+    }
+
+    @Override
+    public ImmutableRTree getSpatialIndex(String dimension)
+    {
+      return null;
+    }
+  };
+  private static final String javaScript = "function(dim1, dim2) { return dim1 === dim2 }";
+  private static final String[] dimensions1 = {"foo", "bar"};
+  private static final String[] dimensions2 = {"foo", "baz"};
+
+  @Test
+  public void testEmpty()
+  {
+    JavaScriptFilter javaScriptFilter = new JavaScriptFilter(
+        dimensions1, javaScript
+    );
+    ImmutableBitmap immutableBitmap = javaScriptFilter.getBitmapIndex(BITMAP_INDEX_SELECTOR);
+    Assert.assertEquals(0, immutableBitmap.size());
+  }
+
+  @Test
+  public void testNormal()
+  {
+    JavaScriptFilter javaScriptFilter = new JavaScriptFilter(
+        dimensions2, javaScript
+    );
+    ImmutableBitmap immutableBitmap = javaScriptFilter.getBitmapIndex(BITMAP_INDEX_SELECTOR);
+    Assert.assertEquals(2, immutableBitmap.size());
+  }
+
+  @Test
+  public void testOr()
+  {
+    Assert.assertEquals(
+        2, Filters.convertDimensionFilters(
+            DimFilters.or(
+                new JavaScriptDimFilter(
+                    null,
+                    dimensions2,
+                    javaScript
+                )
+            )
+        ).getBitmapIndex(BITMAP_INDEX_SELECTOR).size()
+    );
+
+    Assert.assertEquals(
+        2,
+        Filters.convertDimensionFilters(
+            DimFilters.or(
+                new JavaScriptDimFilter(
+                    null,
+                    dimensions1,
+                    javaScript
+                ),
+                new JavaScriptDimFilter(
+                    null,
+                    dimensions2,
+                    javaScript
+                )
+            )
+        ).getBitmapIndex(BITMAP_INDEX_SELECTOR).size()
+    );
+  }
+
+  @Test
+  public void testAnd()
+  {
+    Assert.assertEquals(
+        2, Filters.convertDimensionFilters(
+            DimFilters.or(
+                new JavaScriptDimFilter(
+                    null,
+                    dimensions2,
+                    javaScript
+                )
+            )
+        ).getBitmapIndex(BITMAP_INDEX_SELECTOR).size()
+    );
+
+    Assert.assertEquals(
+        0,
+        Filters.convertDimensionFilters(
+            DimFilters.and(
+                new JavaScriptDimFilter(
+                    null,
+                    dimensions1,
+                    javaScript
+                ),
+                new JavaScriptDimFilter(
+                    null,
+                    dimensions2,
+                    javaScript
+                )
+            )
+        ).getBitmapIndex(BITMAP_INDEX_SELECTOR).size()
+    );
+  }
+
+  /*
+   * commented because complement() of roaring bitmap does not work correctly yet
```

Comment:
this should be fixed with the latest version of roaring
@drcrallen we updated master right?